### DISCUSSION TITLE
Getting camera to record between certain hours of the day

### DISCUSSION THREAD

**Discussion by caden-fluegge** [gopro/labs#854] [General]
Hello - 
I am trying to utilize labs on a GoPro Hero12 to record for one minute every hour between the hours of 07:00 and 19:00.  Eventually the camera will be connected to an external battery for extended deployment times, but I am running tests on the standard Enduro battery to ensure my commands are working.  I have gone through 3 iterations now, each recording for a minute each hour, but not staying within the time constraints specified above.  I've tried playing with the QR code generator, adding my own commands, and working with the blockly.  I'm sure it's something simple I'm missing, just not yet realized.  Below I included my previous attempts (including Blockly where applicable).  I have also looked into similar discussions, and haven't found a solution that worked for me.  I'm not sure where to go next, and would appreciate any feedback!

Thanks in advance!

(Was also attempting to test battery life on different recording modes, so resolution and fps vary between tests)

Attempt 1:

mSoMPRES="18,FAMELTest-1"r180p60fSe1oV0B1D2dV<07:00>19:00!07:00N!S!58E!3540N!R

Attempt 2:

!3540N>07:00<19:00mVr1440p60fW0e1oV0B1D2+!S+!58E+!R!07:00NQ!R

![Attempt2Blockly](https://github.com/gopro/labs/assets/157518687/7a308a8a-2a7d-49ab-a8ef-1fd07d161841)

Attempt 3:

<19:00mVr1440p60fW0e1oV0oB1oD2+!S+!56E~!07:00NQ!3540N!R

![Attempt3Blockly](https://github.com/gopro/labs/assets/157518687/f8261c67-dab5-4538-a002-285436bfa9df)

**Comment by dnewman-gpsw** [comment]
Never do Attempt 1.  You have setting a present inside a repeating function.  Never repeat anything that should be done only once.

Do not set all you video settings with every loop, the camera will remember after setting once. So all the r1400p60fW......... should not be a repeating function.  Also HERO12 doesn't have r1440, it is using r2T (2.7K Tall 4:3.)

Also the reason you code is failing the "added QR Command" is not the correct format to be within a IF statement.  Remove that line to get:
```<19:00!S+!56E~!07:00NQ!3540N!R``` Which will work.

**Reply by caden-fluegge** [reply]
Ok I will give this a go today.  How would you exclude setting a preset/camera settings from the repetition?  Or would it be best to just create a preset which the camera would pull from when starting capture?

Thanks!

**Reply by dnewman-gpsw** [reply]
Easiest is two QR Codes,  or just use a preset.  Or via Blockly use Goto to only loop the parts that need it.

**Reply by caden-fluegge** [reply]
Sweet.  Appreciate it!

**Comment by halesami** [comment]
It sounds like you're interested in setting up a camera to record at specific times. While this isn't directly related to [SASSA grant](https://mysassastatuscheck.com/) information, it's still relevant in terms of managing security or monitoring activities, which can be valuable for individuals receiving grants. However, if you're looking for information specifically related to SASSA grants, our website can provide detailed guidance on eligibility criteria, application processes, and updates regarding grant disbursements. We aim to ensure that individuals have access to accurate and timely information to support their financial well-being.
